.github/workflows/smoke_tests.yaml

```diff
@@ -0,0 +1,33 @@
+# only has to pass for python 3.9
+name: Smoke Tests
+
+on:
+  push:
+    branches:
+      main
+  pull_request:
+    branches:
+      main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v3
+        with:
+          python-version: 3.9
+          cache: 'pip'
+      # Display the Python version being used
+      - name: Display Python version
+        run: python -c "import sys; print(sys.version)"
+      - name: Install Requirements
+        run: |
+          pip install --upgrade pip
+          pip install $(grep -v '^torchdata\|^torchtext\|^torcheval' requirements.txt)
```

Comment:
This is a bit of a hack that I put in to forgo installing these packages in the test environment, as they are not necessary, but I don't love how I accomplished it. Very open to ideas if you have any.

Reply:
One thing I did in the past to solve that is to have two separate requirements files, one for test and one for prod. Also solves the issue of having test packages being unnecessarily added as dependencies for prod.

Reply:
For that, the only thing I don't love is that then we have to maintain two sets of requirement files. We can leave it for now and maybe put a ticket in the backlog to consider it more deeply later.